### PR TITLE
DATACOUCH-356 - Retrieve AsyncViewRow from Observable

### PR DESCRIPTION
**Pull request by rmsantana**
Retrieve AsyncViewRow from Observable instead of using cast

- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATACOUCH).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@rmsantana Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-couchbase&pullRequestId=154)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-data-couchbase&pullRequestId=154) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@rmsantana Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-couchbase&pullRequestId=154)!

File: src/main/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepository.java [hunk 1]
```diff
@@ -251,7 +252,7 @@ protected Flux mapFlux(Observable observable) {
         return mapMono(operations.queryView(query)
                 .map(AsyncViewResult::rows)
                 .flatMap(row -> {
-                    AsyncViewRow asyncViewRow = (AsyncViewRow) row;
+                    AsyncViewRow asyncViewRow = row.toBlocking().single();
```
Comment (mp911de) [inline src/main/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepository.java]: That's a blocking call within a non-blocking flow. It should be rather flat-mapped in the operator before instead enforcing synchronization here.

File: src/test/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import rx.Observable;
+import rx.Observable.OnSubscribe;
+
+/**
+ * JUnit test for SimpleReactiveCouchbaseRepository
+ *
+ * @author Rafael Moreti Santana
+ * @since 3.1.0
+ */
+public class SimpleReactiveCouchbaseRepositoryTest {
```
Comment (mp911de) [inline src/test/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepositoryTest.java]: Nit: Our tests are named `…Tests` and not `…Test`.

File: src/test/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	private SimpleReactiveCouchbaseRepository<String, String> repository;
+
+	@Before
+	public void initMocks() {
+
+		OnSubscribe<AsyncViewResult> onSubscribeResult = subscriber -> {
+
+			OnSubscribe<AsyncViewRow> onSubscribeRow = rowSubscriber -> {
+				try {
+					TimeUnit.SECONDS.sleep(5); // some delay..
```
Comment (mp911de) [inline src/test/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepositoryTest.java]: Why is this `Thread.sleep(…)` here? What's the purpose of it?

File: src/test/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+				} catch (InterruptedException e) {
+					e.printStackTrace();
+				}
+				AsyncViewRow mockCountRow11 = mock(AsyncViewRow.class);
```
Comment (mp911de) [inline src/test/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepositoryTest.java]: Please use meaningful variable names.

File: src/test/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepositoryTest.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+		CouchbaseEntityInformation metadata = mock(CouchbaseEntityInformation.class);
+		when(metadata.getJavaType()).thenReturn(String.class);
+
+		repository = new SimpleReactiveCouchbaseRepository<String, String>(metadata, rxJavaCouchbaseOperations);
+		repository.setViewMetadataProvider(mock(ViewMetadataProvider.class));
+	}
+
+	@Test
+	public void shouldDeleteAll() {
+		repository.deleteAll();
```
Comment (mp911de) [inline src/test/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepositoryTest.java]: There's no assertion.

File: src/test/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
+package org.springframework.data.couchbase.repository.support;
+
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
```
Comment (mp911de) [inline src/test/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepositoryTest.java]: Please format your code as described in our [contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc), see [here for a formatter](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide).

File: src/test/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepositoryTest.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+				rowSubscriber.onCompleted();
+			};
+
+			Observable obs = Observable.create(onSubscribeRow);
+
+			AsyncViewResult asyncViewResult = mock(AsyncViewResult.class);
+			when(asyncViewResult.rows()).thenReturn(obs);
+
+			try {
+				TimeUnit.SECONDS.sleep(5); // some delay..
```
Comment (mp911de) [inline src/test/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepositoryTest.java]: Why is this `Thread.sleep(…)` here? What's the purpose of it?

**Comment by xiao0yy** [comment]
Is there any other plan to fix this issue? I still get this issue when calling deleteAll.

**Comment by davidkelly** [comment]
Closing as we have other PRs in-flight to address this issue

**Pull request closed without merging**